### PR TITLE
feat: New ruleset

### PR DESCRIPTION
**Pull request by Cody-Jian**
### 🤔 This is a ...

- [x] New feature
- [x] Workflow
- [x] Documentation update

### 💡 Background and summary

Introducing new ruleset to protect feature branches
Adding guide file to explain how the ruleset works

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Docs are updated/provided or not needed
- [x] Dev is updated/provided or not needed
- [x] Title is descriptive and succinct for changelog
- [x] Environment variables are updated/provided in deployment slots or not needed

---

### CODE REVIEW

File: .github/workflows/protect-branch.yml
Comment (BlazPocrnja) [inline .github/workflows/protect-branch.yml:1]: I think we can delete this file now -- as its not being used anymore.

File: .github/workflows/repo-rulesets.yml
Comment (BlazPocrnja) [inline .github/workflows/repo-rulesets.yml:1]: Can we rename this into something more descriptive like the other workflow files ... e.g. update-repo-rulesets.yml

File: .github/workflows/repo-rulesets.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+            curl -L \
+              -X $HTTP_METHOD \
+              -H "Accept: application/vnd.github+json" \
+              -H "Authorization: token $ACCESS_TOKEN" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              -d "$UPDATED_FEATURE_RULESET" \
+              "$REPO_URL"
+
+            # - APPLY Azure ruleset
```
Comment (BlazPocrnja) [inline .github/workflows/repo-rulesets.yml]: Should we protect the default branch as well? e.g. master, main...

File: .github/workflows/protect-branch.yml
Comment (Cody-Jian) [reply .github/workflows/protect-branch.yml:1]: File has been removed

File: .github/workflows/repo-rulesets.yml
Comment (Cody-Jian) [reply .github/workflows/repo-rulesets.yml:1]: Renamed workflow file with descriptive name https://github.com/EDBlueEnergy/.github/pull/2/commits/6584a2ed6f772c58a17731b6b0b4161d594d54b1

File: .github/workflows/repo-rulesets.yml (hunk 1 above)
Comment (Cody-Jian) [reply .github/workflows/repo-rulesets.yml]: Automated default branch path added https://github.com/EDBlueEnergy/.github/pull/2/commits/be94ec5b8ca3156fbbbd7887a04aeca974276cb6

**Review by BlazPocrnja — APPROVED** [review]

**Pull request merged**